File: web-server/plugins/slycat-project-wizards/slycat-project-wizards.py
# Copyright (c) 2013, 2018 National Technology and Engineering Solutions of Sandia, LLC . Under the terms of Contract
# DE-NA0003525 with National Technology and Engineering Solutions of Sandia, LLC, the U.S. Government
# retains certain rights in this software.

def register_slycat_plugin(context):
  import os
  context.register_wizard("slycat-create-project", "New Project", require={"action":"create", "context":"global"})
  context.register_wizard("slycat-edit-project", "Edit Project", require={"action":"edit", "context":"global"})
  context.register_wizard("slycat-info-project", "Project Details", require={"action":"info", "context":"global"})
  context.register_wizard("slycat-delete-project", "Delete Project", require={"action":"delete", "context":"project"})
  

File: web-server/plugins/slycat-project-wizards/slycat-project-wizards.js
import ko from "knockout";

import slycat_create_project from 'plugins/slycat-project-wizards/create-ui';
import slycat_edit_project from 'plugins/slycat-project-wizards/edit-ui';
import slycat_info_project from 'plugins/slycat-project-wizards/info-ui';
import slycat_delete_project from 'plugins/slycat-project-wizards/delete-ui';

ko.components.register('slycat-create-project', slycat_create_project);
ko.components.register('slycat-edit-project', slycat_edit_project);
ko.components.register('slycat-info-project', slycat_info_project);
ko.components.register('slycat-delete-project', slycat_delete_project);